snap/pack/pack.go

```diff
@@ -116,6 +117,12 @@ func loadAndValidate(sourceDir string) (*snap.Info, error) {
 	if err := snap.ValidateContainer(snapdir.New(sourceDir), info, logger.Noticef); err != nil {
 		return nil, err
 	}
+
+	if info.SnapType == snap.TypeGadget {
+		if err := gadget.Validate(sourceDir, &gadget.ModelConstraints{Classic: true}); err != nil {
```

Comment:
why Classic: true instead of nil?

Reply:
Switched to nil. Wonder whether we should have a way for figuring out the model constraints somehow at the time when you pack the snap.